api.go

```diff
@@ -155,9 +155,10 @@ func basicAuth(username, password string) string {
 	return base64.StdEncoding.EncodeToString([]byte(auth))
 }
 
-func (c *Client) getJSONWithRetry(
-	method, uri string,
-	body, resp interface{}) error {
+func (c *Client) getJSONWithRetry(method, uri string, body, resp interface{}) error {
+
+	mu.Lock()
```

Comment:
this will have a significant performance impact ,
 if there are 2 create volumes at the same time , they will be serialized here is it not ?